app/src/newdocs/content/canvasModules/Abs-27.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import isEmpty from 'lodash/isEmpty'
+import ReactMarkdown from 'react-markdown'
+
+import { createMdSnippet } from '$newdocs/components/utils'
+import moduleConfig from './Abs-27.json'
+
+const inputs = !isEmpty(moduleConfig.help.inputs)
+    ? createMdSnippet(moduleConfig.help.inputs)
+    : false
+
+const outputs = !isEmpty(moduleConfig.help.outputs)
+    ? createMdSnippet(moduleConfig.help.outputs)
+    : false
+
+const params = !isEmpty(moduleConfig.help.params)
+    ? createMdSnippet(moduleConfig.help.params)
+    : false
+
+export default () => (
```

Comment:
Could these module components be automatically generated, they seem pretty similar?

Reply:
Yep, certainly. Looking into that now. The complexity is in the volume of these importing/exporting modules, some pre scripting is needed.

Tim had some feedback too, summed up:

- Move to 2 files per help-module: 1 `MDX` for the long form `helpText`, 1 `JSX` file that imports the `MDX` file and default exports a module config object (currently the `JSON` file).

- Editor Sidebar would import the `MDX` (long form `helpText` descriptions) from the `MDX` files, and the inputs/outputs/params would come from the exported configuration object from the `JSX` files of each module.

How do you feel about that? I thought it would be better to have a `JSON` source of truth, for simplicity, DRYness and future-proofing but it adds the need for escaping inside the `MD` because it's in a `JSON`, which is a nuisance. I'll see how I go with this refactor, it might turn out cleaner so I will ping when it's ready.